--- compat-gd.spec.old	2012-09-10 14:46:56.000000000 +0800
+++ compat-gd.spec	2017-01-27 10:59:08.000000000 +0800
@@ -1,11 +1,19 @@
 Summary:       A graphics library for quick creation of PNG or JPEG images
+%if 0%{?moffix} == 6
+Name:          compat-gd
+%else
 Name:          gd
+%endif
 Version:       2.0.35
 Release:       11%{?dist}
 Group:         System Environment/Libraries
 License:       MIT
 URL:           http://www.libgd.org/Main_Page
+%if 0%{?moffix} == 6
+Source0:       http://www.libgd.org/releases/gd-%{version}.tar.bz2
+%else
 Source0:       http://www.libgd.org/releases/%{name}-%{version}.tar.bz2
+%endif
 Patch0:        gd-2.0.33-freetype.patch
 Patch3:        gd-2.0.34-multilib.patch
 Patch4:        gd-loop.patch
@@ -29,6 +37,7 @@
 browsers. Note that gd is not a paint program.
 
 
+%if 0%{?moffix} != 6
 %package progs
 Requires:       gd = %{version}-%{release}
 Summary:        Utility programs that use libgd
@@ -50,9 +59,14 @@
 %description devel
 The gd-devel package contains the development libraries and header
 files for gd, a graphics library for creating PNG and JPEG graphics.
+%endif
 
 %prep
+%if 0%{?moffix} == 6
+%setup -q -n gd-%{version}
+%else
 %setup -q
+%endif
 %patch0 -p1 -b .freetype
 %patch3 -p1 -b .mlib
 %patch4 -p1 -b .loop
@@ -70,9 +84,14 @@
 
 %install
 rm -rf $RPM_BUILD_ROOT
+%if 0%{?moffix} == 6
+mkdir -p $RPM_BUILD_ROOT/%{_libdir}/
+cp -a .libs/libgd.so.* $RPM_BUILD_ROOT/%{_libdir}/
+%else
 make install INSTALL='install -p' DESTDIR=$RPM_BUILD_ROOT 
 rm $RPM_BUILD_ROOT/%{_libdir}/libgd.la
 rm $RPM_BUILD_ROOT/%{_libdir}/libgd.a
+%endif
 
 %clean
 rm -rf $RPM_BUILD_ROOT
@@ -87,6 +106,7 @@
 %doc COPYING README-JPEG.TXT index.html NEWS
 %{_libdir}/*.so.*
 
+%if 0%{?moffix} != 6
 %files progs
 %defattr(-,root,root,-)
 %{_bindir}/*
@@ -99,6 +119,7 @@
 %{_includedir}/*
 %{_libdir}/*.so
 %{_libdir}/pkgconfig/gdlib.pc
+%endif
 
 %changelog
 * Mon Sep 10 2012 Honza Horak <hhorak@redhat.com> - 2.0.35-11
