diff -Naur libpsl-0.14.0.old/src/psl-make-dafsa libpsl-0.14.0/src/psl-make-dafsa
--- libpsl-0.14.0.old/src/psl-make-dafsa	2016-07-30 19:56:47.000000000 +0800
+++ libpsl-0.14.0/src/psl-make-dafsa	2017-01-18 11:29:07.000000000 +0800
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2
+#!/usr/bin/env python3
 # Copyright 2014 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE.chromium file.
