--- openvas-gsa.spec.old	2016-09-06 06:20:07.000000000 +0800
+++ openvas-gsa.spec	2017-01-18 16:51:03.000000000 +0800
@@ -48,8 +48,12 @@
 BuildRequires:  gettext
 
 Requires:       logrotate
+%if 0%{?moffix} == 6
+Requires:       texlive-texmf-latex
+%else
 Requires:       texlive-changepage
 Requires:       texlive-comment
+%endif
 
 %if 0%{?rhel} >= 7 || 0%{?fedora} > 15
 BuildRequires:          systemd
@@ -84,7 +88,11 @@
 
 
 %build
+%if 0%{?moffix} == 6
+export CFLAGS="$RPM_OPT_FLAGS -Werror=unused-but-set-variable -lgpg-error -Wno-error=deprecated-declarations -fno-strict-aliasing"
+%else
 export CFLAGS="$RPM_OPT_FLAGS -Werror=unused-but-set-variable -lgpg-error -Wno-error=deprecated-declarations"
+%endif
 %cmake -DLOCALSTATEDIR:PATH=%{_var} -DSYSCONFDIR:PATH=/etc/
 make %{?_smp_mflags} VERBOSE=1
 
