--- gettext.spec.old	2016-12-21 01:42:12.000000000 +0800
+++ gettext.spec	2016-12-24 14:09:54.000000000 +0800
@@ -55,7 +55,6 @@
 Requires(post): info
 Requires(preun): info
 # for F17 UsrMove
-Conflicts: filesystem < 3
 Provides: /bin/gettext
 # exception for bundled gnulib copylib
 Provides: bundled(gnulib)
@@ -183,6 +182,7 @@
 
 install -pm 755 %SOURCE2 ${RPM_BUILD_ROOT}/%{_bindir}/msghack
 install -pm 644 %SOURCE3 ${RPM_BUILD_ROOT}/%{_mandir}/man1/msghack.1
+sed -i -e "s@/usr/bin/python3@/usr/bin/python@g" ${RPM_BUILD_ROOT}/%{_bindir}/msghack
 
 
 # make preloadable_libintl.so executable
