--- automake.spec.old	2016-06-28 21:32:53.000000000 +0800
+++ automake.spec	2017-01-17 14:00:19.000000000 +0800
@@ -52,7 +52,11 @@
 
 # the filtering macros are currently in /etc/rpm/macros.perl:
 BuildRequires: perl-generators
+%if 0%{?moffix} == 6
+BuildRequires: perl-devel
+%else
 BuildRequires: perl-macros
+%endif
 
 # This is needed to allow 'help2man' to run 'automake --help' during build.
 BuildRequires:   perl(Thread::Queue)
@@ -71,6 +75,9 @@
 
 %prep
 %autosetup -p1
+%if 0%{?moffix} == 6
+
+%endif
 autoreconf -iv
 
 file=`find -name config.sub | head -1`
@@ -91,6 +98,9 @@
 
 %install
 make install DESTDIR=%{buildroot}
+%if 0%{?moffix} == 6
+rm -fr %{buildroot}%{_docdir}
+%endif
 
 %check
 # %%global TESTS_FLAGS t/preproc-errmsg t/preproc-basics
@@ -109,7 +119,11 @@
 
 %files
 %doc AUTHORS README THANKS NEWS README.aclocal README.multilib COPYING*
+%if 0%{?moffix} == 6
+%doc doc/amhello-1.0.tar.gz
+%else
 %doc %{_pkgdocdir}/amhello-1.0.tar.gz
+%endif
 %exclude %{_infodir}/dir
 %exclude %{_datadir}/aclocal
 %{_bindir}/*
