--- automake.spec.old	2017-08-15 15:11:56.000000000 +0800
+++ automake.spec	2017-10-27 07:06:28.000000000 +0800
@@ -53,7 +53,11 @@
 
 # the filtering macros are currently in /etc/rpm/macros.perl:
 BuildRequires: perl-generators
+%if 0%{?moffix} == 6
+BuildRequires: perl-devel
+%else
 BuildRequires: perl-macros
+%endif
 
 # This is needed to allow 'help2man' to run 'automake --help' during build.
 BuildRequires:   perl(Thread::Queue)
@@ -72,6 +76,7 @@
 
 %prep
 %autosetup -p1
+
 %if %{with check}
 autoreconf -iv
 %endif
@@ -94,6 +99,9 @@
 
 %install
 make install DESTDIR=%{buildroot}
+%if 0%{?moffix} == 6
+rm -fr %{buildroot}%{_docdir}
+%endif
 
 %check
 # %%global TESTS_FLAGS t/preproc-errmsg t/preproc-basics
@@ -112,7 +120,11 @@
 
 %files
 %doc AUTHORS README THANKS NEWS README.aclocal README.multilib COPYING*
+%if 0%{?moffix} == 6
+%doc doc/amhello-1.0.tar.gz
+%else
 %doc %{_pkgdocdir}/amhello-1.0.tar.gz
+%endif
 %exclude %{_infodir}/dir
 %exclude %{_datadir}/aclocal
 %{_bindir}/*
