--- libevent.spec.old	2012-04-23 22:03:09.000000000 +0800
+++ libevent.spec	2016-12-02 09:06:01.000000000 +0800
@@ -1,4 +1,4 @@
-Name:           libevent
+Name:           compat-libevent
 Version:        1.4.13
 Release:        4%{?dist}
 Summary:        Abstract asynchronous event notification library
@@ -21,39 +21,6 @@
 to call event_dispatch() and can then add or remove events dynamically
 without having to change the event loop.
 
-%package devel
-Summary: Header files, libraries and development documentation for %{name}
-Group: Development/Libraries
-Requires: %{name} = %{version}-%{release}
-Requires: %{name}-headers = %{version}-%{release}
-Requires: %{name}-doc = %{version}-%{release}
-
-%description devel
-This package contains the static libraries documentation for %{name}. 
-If you like to develop programs using %{name}, you will need 
-to install %{name}-devel.
-
-%package doc
-Summary: Development documentation for %{name}
-Group: Development/Libraries
-Requires: %{name}-devel = %{version}-%{release}
-BuildArch: noarch
-
-%description doc
-This package contains the development documentation for %{name}. 
-If you like to develop programs using %{name}-devel, you will 
-need to install %{name}-doc.
-
-%package headers
-Summary: Header file for development  for %{name}
-Group: Development/Libraries
-Requires: %{name}-devel = %{version}-%{release}
-BuildArch: noarch
-
-%description headers
-This package contains the header files for %{name}. If you like to 
-develop programs using %{name}, you will need to install %{name}-devel.
-
 %prep
 %setup -q -n libevent-%{version}-stable
 
@@ -70,24 +37,8 @@
 
 %install
 rm -rf $RPM_BUILD_ROOT
-make DESTDIR=$RPM_BUILD_ROOT install
-rm -f $RPM_BUILD_ROOT%{_libdir}/*.la
-
-mkdir -p $RPM_BUILD_ROOT/%{_docdir}/%{name}-devel-%{version}/html
-(cd doxygen/html; \
-	install *.* $RPM_BUILD_ROOT/%{_docdir}/%{name}-devel-%{version}/html)
-
-mkdir -p $RPM_BUILD_ROOT/%{_docdir}/%{name}-devel-%{version}/latex
-(cd doxygen/latex; \
-	install *.* $RPM_BUILD_ROOT/%{_docdir}/%{name}-devel-%{version}/latex)
-
-mkdir -p $RPM_BUILD_ROOT/%{_docdir}/%{name}-devel-%{version}/man/man3
-(cd doxygen/man/man3; \
-	install *.3 $RPM_BUILD_ROOT/%{_docdir}/%{name}-devel-%{version}/man/man3)
-
-mkdir -p $RPM_BUILD_ROOT/%{_docdir}/%{name}-devel-%{version}/sample
-(cd sample; \
-	install *.c Makefile* $RPM_BUILD_ROOT/%{_docdir}/%{name}-devel-%{version}/sample)
+mkdir -p $RPM_BUILD_ROOT%{_libdir}
+cp -a .libs/libevent*.so.* $RPM_BUILD_ROOT%{_libdir}/
 
 %check
 make verify
@@ -106,35 +57,6 @@
 %{_libdir}/libevent_core-*.so.*
 %{_libdir}/libevent_extra-*.so.*
 
-%files devel
-%defattr(-,root,root,0755)
-%{_libdir}/libevent.so
-%{_libdir}/libevent.a
-%{_libdir}/libevent_core.so
-%{_libdir}/libevent_core.a
-%{_libdir}/libevent_extra.so
-%{_libdir}/libevent_extra.a
-
-%{_bindir}/event_rpcgen.*
-
-%{_mandir}/man3/*
-
-%files doc
-%defattr(-,root,root,0644)
-%{_docdir}/%{name}-devel-%{version}/html/*
-%{_docdir}/%{name}-devel-%{version}/latex/*
-%{_docdir}/%{name}-devel-%{version}/man/man3/*
-%{_docdir}/%{name}-devel-%{version}/sample/*
-
-%files headers
-%{_includedir}/event.h
-%{_includedir}/evdns.h
-%{_includedir}/evhttp.h
-%{_includedir}/evrpc.h
-%{_includedir}/event-config.h
-%{_includedir}/evutil.h
-%defattr(-,root,root,0644)
-
 %changelog
 * Mon Apr 23 2012 Steve Dickson <steved@redhat.com> 1.4.13-4
 - Moved header files into there own rpm (bz 658051)
