--- compat-libevent.spec.old	2012-04-23 22:03:09.000000000 +0800
+++ compat-libevent.spec	2017-01-27 17:38:30.000000000 +0800
@@ -1,4 +1,8 @@
+%if 0%{?moffix} == 6
+Name:           compat-libevent
+%else
 Name:           libevent
+%endif
 Version:        1.4.13
 Release:        4%{?dist}
 Summary:        Abstract asynchronous event notification library
@@ -21,6 +25,7 @@
 to call event_dispatch() and can then add or remove events dynamically
 without having to change the event loop.
 
+%if 0%{?moffix} != 6
 %package devel
 Summary: Header files, libraries and development documentation for %{name}
 Group: Development/Libraries
@@ -53,6 +58,7 @@
 %description headers
 This package contains the header files for %{name}. If you like to 
 develop programs using %{name}, you will need to install %{name}-devel.
+%endif
 
 %prep
 %setup -q -n libevent-%{version}-stable
@@ -70,6 +76,10 @@
 
 %install
 rm -rf $RPM_BUILD_ROOT
+%if 0%{?moffix} == 6
+mkdir -p $RPM_BUILD_ROOT%{_libdir}
+cp -a .libs/libevent*.so.* $RPM_BUILD_ROOT%{_libdir}/
+%else
 make DESTDIR=$RPM_BUILD_ROOT install
 rm -f $RPM_BUILD_ROOT%{_libdir}/*.la
 
@@ -88,6 +98,7 @@
 mkdir -p $RPM_BUILD_ROOT/%{_docdir}/%{name}-devel-%{version}/sample
 (cd sample; \
 	install *.c Makefile* $RPM_BUILD_ROOT/%{_docdir}/%{name}-devel-%{version}/sample)
+%endif
 
 %check
 make verify
@@ -106,6 +117,7 @@
 %{_libdir}/libevent_core-*.so.*
 %{_libdir}/libevent_extra-*.so.*
 
+%if 0%{?moffix} != 6
 %files devel
 %defattr(-,root,root,0755)
 %{_libdir}/libevent.so
@@ -134,6 +146,7 @@
 %{_includedir}/event-config.h
 %{_includedir}/evutil.h
 %defattr(-,root,root,0644)
+%endif
 
 %changelog
 * Mon Apr 23 2012 Steve Dickson <steved@redhat.com> 1.4.13-4
