--- openvas-manager.spec.old	2016-09-06 00:29:13.000000000 +0800
+++ openvas-manager.spec	2017-01-18 16:44:23.000000000 +0800
@@ -26,10 +26,19 @@
 # Replace _BSD_SOURCE and _SVID_SOURCE with _DEFAULT_SOURCE otherwise build fails with Werror
 Patch3:		%{name}-bsdsource.patch
 
+%if 0%{?moffix} == 6
+# Fix: src/manage_sql.c:2656: error: ignoring return value of 'strtol', declared with attribute warn_unused_result
+Patch9:		openvas-manager-6.0.9-unused-result.patch
+%endif
+
 BuildRequires:	openvas-libraries-devel >= 7.0
 BuildRequires:	cmake >= 2.6.0
 BuildRequires:	glib2-devel
+%if 0%{?moffix} == 6
+BuildRequires:	sqlite-devel >= 3.7.15
+%else
 BuildRequires:	sqlite-devel
+%endif
 BuildRequires:	gnutls-devel
 BuildRequires:	libgcrypt-devel
 BuildRequires:	libuuid-devel
@@ -53,6 +62,9 @@
 
 Requires:	logrotate
 Requires:	/usr/bin/xsltproc
+%if 0%{?moffix} == 6
+Requires:	sqlite-libs >= 3.7.15
+%endif
 
 
 %description
@@ -70,6 +82,9 @@
 #%patch0 -p1 -b .notused
 %patch1 -p1 -b .pki
 %patch2 -p1 -b .gpgerror
+%if 0%{?moffix} == 6
+%patch9 -p1 -b .unused-result
+%endif
 
 #%if 0%{?fedora} >= 21
 #%patch3 -p1 -b .bsdsource
@@ -82,7 +97,11 @@
 mv ChangeLog.utf8 ChangeLog
 
 %build
+%if 0%{?moffix} == 6
+export CFLAGS="$RPM_OPT_FLAGS -Werror=unused-but-set-variable -lgpg-error -fno-strict-aliasing"
+%else
 export CFLAGS="$RPM_OPT_FLAGS -Werror=unused-but-set-variable -lgpg-error"
+%endif
 %cmake -DLOCALSTATEDIR:PATH=%{_var}
 make %{?_smp_mflags} VERBOSE=1
 
