--- openvas-manager.spec.old	2016-09-05 22:28:26.000000000 +0800
+++ openvas-manager.spec	2016-12-06 10:18:40.000000000 +0800
@@ -26,6 +26,8 @@
 # Replace _BSD_SOURCE and _SVID_SOURCE with _DEFAULT_SOURCE otherwise build fails with Werror
 Patch3:		%{name}-bsdsource.patch
 
+Patch4:         openvas-manager-gcc44.patch
+
 BuildRequires:	openvas-libraries-devel >= 7.0
 BuildRequires:	cmake >= 2.6.0
 BuildRequires:	glib2-devel
@@ -70,6 +72,7 @@
 #%patch0 -p1 -b .notused
 %patch1 -p1 -b .pki
 %patch2 -p1 -b .gpgerror
+%patch4 -p1 -b .gcc44
 
 #%if 0%{?fedora} >= 21
 #%patch3 -p1 -b .bsdsource
@@ -82,7 +85,6 @@
 mv ChangeLog.utf8 ChangeLog
 
 %build
-export CFLAGS="$RPM_OPT_FLAGS -Werror=unused-but-set-variable -lgpg-error"
 %cmake -DLOCALSTATEDIR:PATH=%{_var}
 make %{?_smp_mflags} VERBOSE=1
 
