--- openvas-manager.spec.old	2016-09-05 22:28:26.000000000 +0800
+++ openvas-manager.spec	2016-12-22 17:28:32.000000000 +0800
@@ -29,7 +29,7 @@
 BuildRequires:	openvas-libraries-devel >= 7.0
 BuildRequires:	cmake >= 2.6.0
 BuildRequires:	glib2-devel
-BuildRequires:	sqlite-devel
+BuildRequires:	sqlite-devel >= 3.7.15
 BuildRequires:	gnutls-devel
 BuildRequires:	libgcrypt-devel
 BuildRequires:	libuuid-devel
@@ -53,6 +53,7 @@
 
 Requires:	logrotate
 Requires:	/usr/bin/xsltproc
+Requires:	sqlite >= 3.7.15
 
 
 %description
@@ -82,7 +83,7 @@
 mv ChangeLog.utf8 ChangeLog
 
 %build
-export CFLAGS="$RPM_OPT_FLAGS -Werror=unused-but-set-variable -lgpg-error"
+export CFLAGS="$RPM_OPT_FLAGS -Werror=unused-but-set-variable -Wno-error=unused-result -lgpg-error"
 %cmake -DLOCALSTATEDIR:PATH=%{_var}
 make %{?_smp_mflags} VERBOSE=1
 
