--- publicsuffix-list.spec.old	2017-01-05 14:19:40.000000000 +0800
+++ publicsuffix-list.spec	2017-01-18 10:53:00.000000000 +0800
@@ -14,7 +14,11 @@
 
 BuildArch:      noarch
 
+%if 0%{?moffix} == 6
+BuildRequires:  python34
+%else
 BuildRequires:  python3
+%endif
 
 
 %description
@@ -47,7 +51,11 @@
 
 
 %build
+%if 0%{?moffix} == 6
+LC_CTYPE=en_US.UTF-8 \
+%else
 LC_CTYPE=C.UTF-8 \
+%endif
 python3 psl-make-dafsa --output-format=binary \
   public_suffix_list.dat public_suffix_list.dafsa
 
