--- gnutls.spec.old	2013-05-31 23:16:38.000000000 +0800
+++ gnutls.spec	2017-01-17 21:35:36.000000000 +0800
@@ -29,6 +29,9 @@
 
 BuildRoot:  %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
 Requires: libgcrypt >= 1.2.2
+%if 0%{?moffix} == 6
+Requires: libtasn1 >= 2.14
+%endif
 # Wildcard bundling exception https://fedorahosted.org/fpc/ticket/174
 Provides: bundled(gnulib) = 20120301
 
