--- openvas-libraries.spec.old	2016-09-05 19:40:17.000000000 +0800
+++ openvas-libraries.spec	2016-12-22 17:24:43.000000000 +0800
@@ -16,10 +16,14 @@
 
 #Fix fo newer version of libssh
 Patch10:	openvas-libraries-libssh.patch
+#Fix for unused const variable in nasl/iconv.c
+Patch11:	openvas-libraries-8.0.8-unused-const-variable.patch
+#Fix for misleading-indentation in nasl/nasl_builtin_synscan.c
+Patch12:	openvas-libraries-8.0.8-misleading-indentation.patch
 
 BuildRoot:	%{_tmppath}/%{name}-%{version}-%{release}-root
 Obsoletes:	openvas-libnasl
-BuildRequires:	glib2-devel
+BuildRequires:	glib2-devel >= 2.30
 BuildRequires:	libgcrypt-devel
 BuildRequires:	gnutls-devel >= 2.12.10
 BuildRequires:	libpcap-devel
@@ -34,6 +38,8 @@
 BuildRequires:	openldap-devel
 BuildRequires:	libssh-devel
 BuildRequires:	hiredis-devel
+Requires:	gnutls >= 2.12.10
+Requires:	glib2 >= 2.30
 
 
 %description
@@ -70,6 +76,10 @@
 
 #Fix for newer version of the libssh
 %patch10 -p 1 -b .libssh
+#Fix for unused const variable in nasl/iconv.c
+%patch11 -p 2 -b .unused-const-variable
+#Fix for misleading-indentation in nasl/nasl_builtin_synscan.c
+%patch12 -p 2 -b .misleading-indentation
 
 #Fix codepage of the Changelog
 iconv -f LATIN1 -t UTF8 < ChangeLog > ChangeLog1
