--- autoconf.spec.old	2017-07-26 11:36:18.000000000 +0800
+++ autoconf.spec	2017-10-27 06:44:31.000000000 +0800
@@ -9,6 +9,10 @@
 URL:        http://www.gnu.org/software/autoconf/
 
 Patch1:     autoconf-2.69-perl-5.22-autoscan.patch
+%if 0%{?moffix} == 6
+Patch2:     autoconf-2.69-test-configure-directories.patch
+Patch3:     autoconf-2.69-test-libtool.patch
+%endif
 
 BuildArch:  noarch
 
@@ -19,11 +23,17 @@
 # m4 >= 1.4.6 is required, >= 1.4.14 is recommended:
 BuildRequires:      m4 >= 1.4.14
 Requires:           m4 >= 1.4.14
+%if 0%{?moffix} != 6
 Requires:           emacs-filesystem
+%endif
 BuildRequires:      emacs
 # the filtering macros are currently in /etc/rpm/macros.perl:
 BuildRequires:      perl-generators
+%if 0%{?moffix} == 6
+BuildRequires:      perl-devel
+%else
 BuildRequires:      perl-macros
+%endif
 BuildRequires:      perl(Data::Dumper)
 # from f19, Text::ParseWords is not the part of 'perl' package
 BuildRequires:      perl(Text::ParseWords)
