--- autoconf.spec.old	2016-07-29 15:09:15.000000000 +0800
+++ autoconf.spec	2016-12-08 17:09:45.000000000 +0800
@@ -19,11 +19,9 @@
 # m4 >= 1.4.6 is required, >= 1.4.14 is recommended:
 BuildRequires:      m4 >= 1.4.14
 Requires:           m4 >= 1.4.14
-Requires:           emacs-filesystem
 BuildRequires:      emacs
 # the filtering macros are currently in /etc/rpm/macros.perl:
 BuildRequires:      perl-generators
-BuildRequires:      perl-macros
 BuildRequires:      perl(Data::Dumper)
 # from f19, Text::ParseWords is not the part of 'perl' package
 BuildRequires:      perl(Text::ParseWords)
