--- httpd.spec.old	2016-12-22 23:59:50.000000000 +0800
+++ httpd.spec	2017-01-04 16:19:11.000000000 +0800
@@ -13,6 +13,7 @@
 Source0: http://www.apache.org/dist/httpd/httpd-%{version}.tar.bz2
 Source1: index.html
 Source2: httpd.logrotate
+Source3: httpd.init
 Source4: httpd-ssl-pass-dialog
 Source5: httpd.tmpfiles
 Source6: httpd.service
@@ -42,6 +43,7 @@
 Source31: README.confmod
 Source40: htcacheclean.service
 Source41: htcacheclean.sysconf
+Source42: htcacheclean.init
 # build/scripts patches
 Patch1: httpd-2.4.1-apctl.patch
 Patch2: httpd-2.4.9-apxs.patch
@@ -73,8 +75,8 @@
 BuildRequires: autoconf, perl, perl-generators, pkgconfig, findutils, xmlto
 BuildRequires: zlib-devel, libselinux-devel, lua-devel
 BuildRequires: apr-devel >= 1.5.0, apr-util-devel >= 1.5.0, pcre-devel >= 5.0
-BuildRequires: systemd-devel, libnghttp2-devel
-Requires: /etc/mime.types, system-logos-httpd
+BuildRequires: libnghttp2-devel
+Requires: /etc/mime.types, redhat-logos
 Obsoletes: httpd-suexec
 Provides: webserver
 Provides: mod_dav = %{version}-%{release}, httpd-suexec = %{version}-%{release}
@@ -83,9 +85,7 @@
 Requires: httpd-filesystem = %{version}-%{release}
 Requires: nghttp2 >= 1.5.0
 Requires(pre): httpd-filesystem
-Requires(preun): systemd-units
-Requires(postun): systemd-units
-Requires(post): systemd-units
+Requires(post): chkconfig
 Conflicts: apr < 1.5.0-1
 
 %description
@@ -144,7 +144,7 @@
 Summary: SSL/TLS module for the Apache HTTP Server
 Epoch: 1
 BuildRequires: openssl-devel
-Requires(post): openssl, /bin/cat, hostname
+Requires(post): openssl, /bin/cat, /bin/hostname
 Requires(pre): httpd-filesystem
 Requires: httpd = 0:%{version}-%{release}, httpd-mmn = %{mmnisa}
 Obsoletes: stronghold-mod_ssl
@@ -193,17 +193,14 @@
 %patch2 -p1 -b .apxs
 %patch3 -p1 -b .deplibs
 %patch5 -p1 -b .layout
-%patch6 -p1 -b .apctlsystemd
+sed -i -e "s@/run/httpd@/var/run/httpd@g" config.layout
 %patch7 -p1 -b .openssl11
 
-%patch19 -p1 -b .detectsystemd
-
 %patch23 -p1 -b .export
 %patch24 -p1 -b .corelimit
 %patch25 -p1 -b .selinux
 %patch26 -p1 -b .r1337344+
 %patch27 -p1 -b .icons
-%patch29 -p1 -b .systemd
 %patch30 -p1 -b .cachehardmax
 %patch31 -p1 -b .sslmultiproxy
 %patch34 -p1 -b .socketactivation
@@ -288,12 +285,32 @@
 
 make DESTDIR=$RPM_BUILD_ROOT install
 
-# Install systemd service files
-mkdir -p $RPM_BUILD_ROOT%{_unitdir}
-for s in httpd.service htcacheclean.service httpd.socket; do
-  install -p -m 644 $RPM_SOURCE_DIR/${s} \
-                    $RPM_BUILD_ROOT%{_unitdir}/${s}
-done
+# install SYSV init stuff
+mkdir -p $RPM_BUILD_ROOT/etc/rc.d/init.d
+install -m755 $RPM_SOURCE_DIR/httpd.init \
+        $RPM_BUILD_ROOT/etc/rc.d/init.d/httpd
+install -m755 $RPM_SOURCE_DIR/htcacheclean.init \
+        $RPM_BUILD_ROOT/etc/rc.d/init.d/htcacheclean
+
+# Change the httpd.init paths
+sed -i 's|\$sbindir|%{_sbindir}|' \
+    $RPM_BUILD_ROOT/etc/rc.d/init.d/httpd
+sed -i 's|\$sysconfdir|%{_sysconfdir}|' \
+    $RPM_BUILD_ROOT/etc/rc.d/init.d/httpd
+sed -i 's|\$localstatedir|%{_localstatedir}|' \
+    $RPM_BUILD_ROOT/etc/rc.d/init.d/httpd
+sed -i 's|\$sclscripts|%{?_scl_scripts}|' \
+    $RPM_BUILD_ROOT/etc/rc.d/init.d/httpd
+
+# Change the htcacheclean.init paths
+sed -i 's|\$sbindir|%{_sbindir}|' \
+    $RPM_BUILD_ROOT/etc/rc.d/init.d/htcacheclean
+sed -i 's|\$sysconfdir|%{_sysconfdir}|' \
+    $RPM_BUILD_ROOT/etc/rc.d/init.d/htcacheclean
+sed -i 's|\$localstatedir|%{_localstatedir}|' \
+    $RPM_BUILD_ROOT/etc/rc.d/init.d/htcacheclean
+sed -i 's|\$sclscripts|%{?_scl_scripts}|' \
+    $RPM_BUILD_ROOT/etc/rc.d/init.d/htcacheclean
 
 # install conf file/directory
 mkdir $RPM_BUILD_ROOT%{_sysconfdir}/httpd/conf.d \
@@ -304,20 +321,11 @@
     $RPM_BUILD_ROOT%{_sysconfdir}/httpd/conf.modules.d/README
 for f in 00-base.conf 00-mpm.conf 00-lua.conf 01-cgi.conf 00-dav.conf \
          00-proxy.conf 00-ssl.conf 01-ldap.conf 00-proxyhtml.conf \
-         01-ldap.conf 00-systemd.conf 01-session.conf 00-optional.conf; do
+         01-ldap.conf 01-session.conf 00-optional.conf; do
   install -m 644 -p $RPM_SOURCE_DIR/$f \
         $RPM_BUILD_ROOT%{_sysconfdir}/httpd/conf.modules.d/$f
 done
 
-# install systemd override drop directory
-# Web application packages can drop snippets into this location if
-# they need ExecStart[pre|post].
-mkdir $RPM_BUILD_ROOT%{_unitdir}/httpd.service.d
-mkdir $RPM_BUILD_ROOT%{_unitdir}/httpd.socket.d
-
-install -m 644 -p $RPM_SOURCE_DIR/10-listen443.conf \
-      $RPM_BUILD_ROOT%{_unitdir}/httpd.socket.d/10-listen443.conf
-
 for f in welcome.conf ssl.conf manual.conf userdir.conf; do
   install -m 644 -p $RPM_SOURCE_DIR/$f \
         $RPM_BUILD_ROOT%{_sysconfdir}/httpd/conf.d/$f
@@ -340,14 +348,9 @@
 install -m 644 -p $RPM_SOURCE_DIR/htcacheclean.sysconf \
    $RPM_BUILD_ROOT%{_sysconfdir}/sysconfig/htcacheclean
 
-# tmpfiles.d configuration
-mkdir -p $RPM_BUILD_ROOT%{_prefix}/lib/tmpfiles.d 
-install -m 644 -p $RPM_SOURCE_DIR/httpd.tmpfiles \
-   $RPM_BUILD_ROOT%{_prefix}/lib/tmpfiles.d/httpd.conf
-
 # Other directories
 mkdir -p $RPM_BUILD_ROOT%{_localstatedir}/lib/dav \
-         $RPM_BUILD_ROOT/run/httpd/htcacheclean
+         $RPM_BUILD_ROOT/var/run/httpd/htcacheclean
 
 # Substitute in defaults which are usually done (badly) by "make install"
 sed -i \
@@ -365,8 +368,8 @@
 
 # Make the MMN accessible to module packages
 echo %{mmnisa} > $RPM_BUILD_ROOT%{_includedir}/httpd/.mmn
-mkdir -p $RPM_BUILD_ROOT%{_rpmconfigdir}/macros.d
-cat > $RPM_BUILD_ROOT%{_rpmconfigdir}/macros.d/macros.httpd <<EOF
+mkdir -p $RPM_BUILD_ROOT/etc/rpm
+cat > $RPM_BUILD_ROOT/etc/rpm/macros.httpd <<EOF
 %%_httpd_mmn %{mmnisa}
 %%_httpd_apxs %%{_bindir}/apxs
 %%_httpd_modconfdir %%{_sysconfdir}/httpd/conf.modules.d
@@ -406,7 +409,7 @@
 
 # symlinks for /etc/httpd
 ln -s ../..%{_localstatedir}/log/httpd $RPM_BUILD_ROOT/etc/httpd/logs
-ln -s /run/httpd $RPM_BUILD_ROOT/etc/httpd/run
+ln -s /var/run/httpd $RPM_BUILD_ROOT/etc/httpd/run
 ln -s ../..%{_libdir}/httpd/modules $RPM_BUILD_ROOT/etc/httpd/modules
 
 # install http-ssl-pass-dialog
@@ -414,13 +417,6 @@
 install -m755 $RPM_SOURCE_DIR/httpd-ssl-pass-dialog \
 	$RPM_BUILD_ROOT%{_libexecdir}/httpd-ssl-pass-dialog
 
-# Install action scripts
-mkdir -p $RPM_BUILD_ROOT%{_libexecdir}/initscripts/legacy-actions/httpd
-for f in graceful configtest; do
-    install -p -m 755 $RPM_SOURCE_DIR/action-${f}.sh \
-            $RPM_BUILD_ROOT%{_libexecdir}/initscripts/legacy-actions/httpd/${f}
-done
-
 # Install logrotate config
 mkdir -p $RPM_BUILD_ROOT/etc/logrotate.d
 install -m 644 -p $RPM_SOURCE_DIR/httpd.logrotate \
@@ -432,7 +428,7 @@
     -e "s|/usr/local/apache2/conf/magic|/etc/httpd/conf/magic|" \
     -e "s|/usr/local/apache2/logs/error_log|/var/log/httpd/error_log|" \
     -e "s|/usr/local/apache2/logs/access_log|/var/log/httpd/access_log|" \
-    -e "s|/usr/local/apache2/logs/httpd.pid|/run/httpd/httpd.pid|" \
+    -e "s|/usr/local/apache2/logs/httpd.pid|/var/run/httpd/httpd.pid|" \
     -e "s|/usr/local/apache2|/etc/httpd|" < docs/man/httpd.8 \
   > $RPM_BUILD_ROOT%{_mandir}/man8/httpd.8
 
@@ -466,29 +462,22 @@
 exit 0
 
 %post
-%systemd_post httpd.service htcacheclean.service httpd.socket
+# Register the httpd service
+/sbin/chkconfig --add httpd
+/sbin/chkconfig --add htcacheclean
 
 %preun
-%systemd_preun httpd.service htcacheclean.service httpd.socket
+if [ $1 = 0 ]; then
+        /sbin/service httpd stop > /dev/null 2>&1 || :
+        /sbin/chkconfig --del httpd || :
+        /sbin/service htcacheclean stop > /dev/null 2>&1 || :
+        /sbin/chkconfig --del htcacheclean || :
+fi
 
 %postun
-%systemd_postun
-
-# Trigger for conversion from SysV, per guidelines at:
-# https://fedoraproject.org/wiki/Packaging:ScriptletSnippets#Systemd
-%triggerun -- httpd < 2.2.21-5
-# Save the current service runlevel info
-# User must manually run systemd-sysv-convert --apply httpd
-# to migrate them to systemd targets
-/usr/bin/systemd-sysv-convert --save httpd.service >/dev/null 2>&1 ||:
-
-# Run these because the SysV package being removed won't do them
-/sbin/chkconfig --del httpd >/dev/null 2>&1 || :
+/sbin/service httpd condrestart >/dev/null 2>&1 || :
 
 %posttrans
-test -f /etc/sysconfig/httpd-disable-posttrans || \
-  /bin/systemctl try-restart httpd.service htcacheclean.service >/dev/null 2>&1 || :
-
 %define sslcert %{_sysconfdir}/pki/tls/certs/localhost.crt
 %define sslkey %{_sysconfdir}/pki/tls/private/localhost.key
 
@@ -577,10 +566,6 @@
 %exclude %{_sysconfdir}/httpd/conf.modules.d/01-session.conf
 
 %config(noreplace) %{_sysconfdir}/sysconfig/ht*
-%{_prefix}/lib/tmpfiles.d/httpd.conf
-
-%dir %{_libexecdir}/initscripts/legacy-actions/httpd
-%{_libexecdir}/initscripts/legacy-actions/httpd/*
 
 %{_sbindir}/ht*
 %{_sbindir}/fcgistarter
@@ -607,8 +592,8 @@
 %{contentdir}/error/include/*.html
 %{contentdir}/noindex/index.html
 
-%attr(0710,root,apache) %dir /run/httpd
-%attr(0700,apache,apache) %dir /run/httpd/htcacheclean
+%attr(0710,root,apache) %dir /var/run/httpd
+%attr(0700,apache,apache) %dir /var/run/httpd/htcacheclean
 %attr(0700,root,root) %dir %{_localstatedir}/log/httpd
 %attr(0700,apache,apache) %dir %{_localstatedir}/lib/dav
 %attr(0700,apache,apache) %dir %{_localstatedir}/cache/httpd
@@ -616,10 +601,8 @@
 
 %{_mandir}/man8/*
 
-%{_unitdir}/*.service
-%{_unitdir}/*.socket
-%attr(755,root,root) %dir %{_unitdir}/httpd.service.d
-%attr(755,root,root) %dir %{_unitdir}/httpd.socket.d
+/etc/rc.d/init.d/httpd
+/etc/rc.d/init.d/htcacheclean
 
 %files filesystem
 %dir %{_sysconfdir}/httpd
@@ -651,7 +634,6 @@
 %config(noreplace) %{_sysconfdir}/httpd/conf.d/ssl.conf
 %attr(0700,apache,root) %dir %{_localstatedir}/cache/httpd/ssl
 %{_libexecdir}/httpd-ssl-pass-dialog
-%{_unitdir}/httpd.socket.d/10-listen443.conf
 
 %files -n mod_proxy_html
 %defattr(-,root,root)
@@ -678,7 +660,7 @@
 %dir %{_libdir}/httpd/build
 %{_libdir}/httpd/build/*.mk
 %{_libdir}/httpd/build/*.sh
-%{_rpmconfigdir}/macros.d/macros.httpd
+/etc/rpm/macros.httpd
 
 %changelog
 * Thu Dec 22 2016 Luboš Uhliarik <luhliari@redhat.com> - 2.4.25-2
