--- mariadb.spec.old	2017-02-16 00:07:02.000000000 +0800
+++ mariadb.spec	2017-02-21 14:00:21.000000000 +0800
@@ -231,6 +231,9 @@
 Requires:         %{name}-common%{?_isa} = %{sameevr}
 # Explicit EVR requirement for -libs is needed for RHBZ#1406320
 Requires:         %{name}-libs%{?_isa} = %{sameevr}
+%if 0%{?moffix} == 6
+Requires:         %{name}-config%{?_isa} = %{sameevr}
+%endif
 
 %if %{with mysql_names}
 Provides:         mysql = %{sameevr}
@@ -239,7 +242,9 @@
 Provides:         mysql-compat-client%{?_isa} = %{sameevr}
 %endif
 
+%if 0%{?moffix} != 6
 Suggests:         %{name}-server%{?_isa} = %{sameevr}
+%endif
 
 # MySQL (with caps) is upstream's spelling of their own RPMs for mysql
 %{?obsoleted_mysql_case_evr:Obsoletes: MySQL < %{obsoleted_mysql_case_evr}}
@@ -276,6 +281,9 @@
 Summary:          The shared libraries required for MariaDB/MySQL clients
 Group:            Applications/Databases
 Requires:         %{name}-common%{?_isa} = %{sameevr}
+%if 0%{?moffix} == 6
+Requires:         %{name}-config%{?_isa} = %{sameevr}
+%endif
 %if %{with mysql_names}
 Provides:         mysql-libs = %{sameevr}
 Provides:         mysql-libs%{?_isa} = %{sameevr}
@@ -378,7 +386,11 @@
 %endif
 Requires:         %{name}-common%{?_isa} = %{sameevr}
 Requires:         %{name}-errmsg%{?_isa} = %{sameevr}
+%if 0%{?moffix} == 6
+Requires:         %{name}-server-utils%{?_isa} = %{sameevr}
+%else
 Recommends:       %{name}-server-utils%{?_isa} = %{sameevr}
+%endif
 Requires:         %{_sysconfdir}/my.cnf
 Requires:         %{_sysconfdir}/my.cnf.d
 Requires:         sh-utils
@@ -616,6 +628,9 @@
 %ifarch ppc ppc64 ppc64p7 s390 s390x
 cat %{SOURCE52} | tee -a mysql-test/rh-skipped-tests.list
 %endif
+%if 0%{?moffix} == 6
+echo "plugins.feedback_plugin_load                    : failed" >> mysql-test/rh-skipped-tests.list
+%endif
 
 cp %{SOURCE2} %{SOURCE3} %{SOURCE10} %{SOURCE11} %{SOURCE12} %{SOURCE13} \
    %{SOURCE14} %{SOURCE15} %{SOURCE16} %{SOURCE17} %{SOURCE18} %{SOURCE19} \
@@ -1196,7 +1211,9 @@
 %{_bindir}/aria_ftdump
 %{_bindir}/aria_pack
 %{_bindir}/aria_read_log
+%if 0%{?moffix} != 6
 %{_bindir}/mariadb-service-convert
+%endif
 %{_bindir}/myisamchk
 %{_bindir}/myisam_ftdump
 %{_bindir}/myisamlog
@@ -1282,11 +1299,13 @@
 %{_datadir}/%{pkg_name}/policy/selinux/README
 %{_datadir}/%{pkg_name}/policy/selinux/mariadb-server.*
 %{_datadir}/%{pkg_name}/policy/selinux/mariadb.*
+%if 0%{?moffix} != 6
 %{_datadir}/%{pkg_name}/systemd/mariadb.service
 # mariadb@ is installed only when we have cmake newer than 3.3
 %if 0%{?fedora} > 22 || 0%{?rhel} > 7
 %{_datadir}/%{pkg_name}/systemd/mariadb@.service
 %endif
+%endif
 
 %{daemondir}/%{daemon_name}*
 %{_libexecdir}/mysql-prepare-db-dir
