--- mariadb.spec.old	2016-12-17 07:37:46.000000000 +0800
+++ mariadb.spec	2016-12-29 14:02:39.000000000 +0800
@@ -25,7 +25,7 @@
 # variable tokudb allows to build with TokuDB storage engine
 # Temporarily disabled in F21+ for https://mariadb.atlassian.net/browse/MDEV-6446
 %ifarch x86_64
-%bcond_without tokudb
+%bcond_with tokudb
 %else
 %bcond_with tokudb
 %endif
@@ -605,6 +605,8 @@
 cat %{SOURCE52} | tee -a mysql-test/rh-skipped-tests.list
 %endif
 
+echo "plugins.feedback_plugin_load                    : failed" >> mysql-test/rh-skipped-tests.list
+
 cp %{SOURCE2} %{SOURCE3} %{SOURCE10} %{SOURCE11} %{SOURCE12} %{SOURCE13} \
    %{SOURCE14} %{SOURCE15} %{SOURCE16} %{SOURCE17} %{SOURCE18} %{SOURCE19} \
    %{SOURCE70} scripts
@@ -759,7 +761,7 @@
 # but that's pretty wacko --- see also %%{name}-file-contents.patch)
 install -p -m 644 Docs/INFO_SRC %{buildroot}%{_libdir}/mysql/
 install -p -m 644 Docs/INFO_BIN %{buildroot}%{_libdir}/mysql/
-rm -rf %{buildroot}%{_pkgdocdir}/MariaDB-server-%{version}/
+rm -rf %{buildroot}%{_docdir}/%{_pkgdocdirname}/MariaDB-server-%{version}/
 
 mkdir -p %{buildroot}%{logfiledir}
 chmod 0750 %{buildroot}%{logfiledir}
@@ -1163,7 +1165,9 @@
 %{_bindir}/aria_ftdump
 %{_bindir}/aria_pack
 %{_bindir}/aria_read_log
+%if %{with init_systemd}
 %{_bindir}/mariadb-service-convert
+%endif
 %{_bindir}/myisamchk
 %{_bindir}/myisam_ftdump
 %{_bindir}/myisamlog
@@ -1265,11 +1269,13 @@
 %{_datadir}/%{pkg_name}/policy/apparmor/usr.sbin.mysqld*
 %{_datadir}/%{pkg_name}/policy/selinux/README
 %{_datadir}/%{pkg_name}/policy/selinux/mariadb-server.*
+%if %{with init_systemd}
 %{_datadir}/%{pkg_name}/systemd/mariadb.service
 # mariadb@ is installed only when we have cmake newer than 3.3
 %if 0%{?fedora} > 22 || 0%{?rhel} > 7
 %{_datadir}/%{pkg_name}/systemd/mariadb@.service
 %endif
+%endif
 
 %{daemondir}/%{daemon_name}*
 %{_libexecdir}/mysql-prepare-db-dir
