--- compat-libzip.spec.old	2009-12-04 21:45:40.000000000 +0800
+++ compat-libzip.spec	2017-01-27 20:03:19.000000000 +0800
@@ -1,7 +1,11 @@
 # Fedora Review Request #393041
 # https://bugzilla.redhat.com/show_bug.cgi?id=393041
 
+%if 0%{?moffix} == 6
+Name:           compat-libzip
+%else
 Name:           libzip
+%endif
 Version:        0.9
 Release:        3.1%{?dist}
 Summary:        C library for reading, creating, and modifying zip archives
@@ -9,7 +13,11 @@
 Group:          System Environment/Libraries
 License:        BSD
 URL:            http://www.nih.at/libzip/index.html
+%if 0%{?moffix} == 6
+Source0:        http://www.nih.at/libzip/libzip-%{version}.tar.gz
+%else
 Source0:        http://www.nih.at/libzip/%{name}-%{version}.tar.gz
+%endif
 BuildRoot:      %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
 
 BuildRequires:  automake libtool
@@ -22,6 +30,7 @@
 other zip archives. Changes made without closing the archive can be reverted. 
 The API is documented by man pages.
 
+%if 0%{?moffix} != 6
 %package devel
 Summary: Development files for %{name}
 Group: Development/Libraries
@@ -31,10 +40,15 @@
 %description devel
 The %{name}-devel package contains libraries and header files for
 developing applications that use %{name}.
+%endif
 
 
 %prep
+%if 0%{?moffix} == 6
+%setup -q -n libzip-%{version}
+%else
 %setup -q
+%endif
 
 # Avoid lib64 rpaths (FIXME: recheck this on newer releases)
 #if "%{_libdir}" != "/usr/lib"
@@ -50,8 +64,13 @@
 
 %install
 rm -rf $RPM_BUILD_ROOT
+%if 0%{?moffix} == 6
+mkdir -p $RPM_BUILD_ROOT%{_libdir}
+cp -a lib/.libs/libzip.so.* $RPM_BUILD_ROOT%{_libdir}/
+%else
 make install DESTDIR=$RPM_BUILD_ROOT INSTALL='install -p'
 find $RPM_BUILD_ROOT -name '*.la' -exec rm -f {} ';'
+%endif
 
 
 %clean
@@ -66,18 +85,24 @@
 %files
 %defattr(-,root,root,-)
 %doc AUTHORS NEWS README THANKS TODO
+%if 0%{?moffix} == 6
+%{_libdir}/libzip.so.1*
+%else
 %{_bindir}/zipcmp
 %{_bindir}/zipmerge
 %{_bindir}/ziptorrent
 %{_libdir}/libzip.so.1*
 %{_mandir}/man1/*zip*
+%endif
 
+%if 0%{?moffix} != 6
 %files devel
 %defattr(-,root,root,-)
 %{_includedir}/zip.h
 %{_libdir}/libzip.so
 %{_libdir}/pkgconfig/libzip.pc
 %{_mandir}/man3/*zip*
+%endif
 
 
 %changelog
