--- bind.spec.old	2016-12-20 01:42:36.000000000 +0800
+++ bind.spec	2017-01-03 14:01:57.000000000 +0800
@@ -33,7 +33,9 @@
 #
 Source:   https://ftp.isc.org/isc/bind9/%{VERSION}/bind-%{VERSION}.tar.gz
 Source1:  named.sysconfig
+Source2:  named.init
 Source3:  named.logrotate
+Source4:  named.NetworkManager
 Source7:  bind-9.3.1rc1-sdb_tools-Makefile.in
 Source8:  dnszone.schema
 Source12: README.sdb_pgsql
@@ -93,13 +95,12 @@
 Patch73: bind-99-libidn.patch
 
 
-Requires(post):   systemd
-Requires(preun):  systemd
-Requires(postun): systemd
 Requires:       coreutils
+Requires(post): grep, chkconfig
 Requires(pre):  shadow-utils
-Requires:	python3-ply
-BuildRequires:	python3-ply
+Requires(preun):chkconfig
+Requires:	python34-ply
+BuildRequires:	python34-ply
 Requires:       bind-libs%{?_isa} = %{epoch}:%{version}-%{release}
 Obsoletes:      bind-config < 30:9.3.2-34.fc6
 Provides:       bind-config = 30:9.3.2-34.fc6
@@ -109,12 +110,11 @@
 Provides:       dnssec-conf = 1.27-1
 BuildRequires:  openssl-devel, libtool, autoconf, pkgconfig, libcap-devel
 BuildRequires:  libidn-devel, libxml2-devel, GeoIP-devel
-BuildRequires:  systemd
 # needed for %%{__python3} macro
-BuildRequires:  python3-devel
+BuildRequires:  python34-devel
 %if %{SDB}
 BuildRequires:  openldap-devel, postgresql-devel, sqlite-devel, mysql-devel
-BuildRequires:  libdb-devel
+BuildRequires:  db4-devel
 %endif
 %if %{test}
 BuildRequires:  net-tools
@@ -136,7 +136,6 @@
 %package pkcs11
 Summary: Bind with native PKCS#11 functionality for crypto
 Group:   System Environment/Daemons
-Requires: systemd
 Requires: bind%{?_isa} = %{epoch}:%{version}-%{release}
 Requires: bind-pkcs11-libs%{?_isa} = %{epoch}:%{version}-%{release}
 
@@ -181,7 +180,6 @@
 %package sdb
 Summary: BIND server with database backends and DLZ support
 Group:   System Environment/Daemons
-Requires: systemd
 Requires: bind%{?_isa} = %{epoch}:%{version}-%{release}
 Requires: bind-libs%{?_isa} = %{epoch}:%{version}-%{release}
 
@@ -276,29 +274,14 @@
 chroot(2) jail for the named(8) program from the BIND package.
 Based on the code from Jan "Yenya" Kasprzak <kas@fi.muni.cz>
 
-%if %{SDB}
-%package sdb-chroot
-Summary:        A chroot runtime environment for the ISC BIND DNS server, named-sdb(8)
-Group:          System Environment/Daemons
-Prefix:         %{chroot_sdb_prefix}
-# grep is required due to setup-named-chroot.sh script
-Requires:       grep
-Requires:       bind-sdb%{?_isa} = %{epoch}:%{version}-%{release}
-
-%description sdb-chroot
-This package contains a tree of files which can be used as a
-chroot(2) jail for the named-sdb(8) program from the BIND package.
-Based on the code from Jan "Yenya" Kasprzak <kas@fi.muni.cz>
-%endif
-
-%package -n python3-bind
+%package -n python34-bind
 Summary:	A module allowing rndc commands to be sent from Python programs
 Group:		Applications/System
 Requires:	bind-license = %{epoch}:%{version}-%{release}
 Requires: 	python3
-%{?python_provide:%python_provide python3-bind}
+%{?python_provide:%python_provide python34-bind}
 
-%description -n python3-bind
+%description -n python34-bind
 This package provides a module which allows commands to be sent to rndc directly from Python programs.
 
 %prep
@@ -449,23 +432,18 @@
 rm -rf ${RPM_BUILD_ROOT}
 
 # Build directory hierarchy
-mkdir -p ${RPM_BUILD_ROOT}/etc/logrotate.d
+mkdir -p ${RPM_BUILD_ROOT}/etc/{rc.d/init.d,logrotate.d,NetworkManager/dispatcher.d}
 mkdir -p ${RPM_BUILD_ROOT}%{_libdir}/bind
 mkdir -p ${RPM_BUILD_ROOT}/var/named/{slaves,data,dynamic}
 mkdir -p ${RPM_BUILD_ROOT}%{_mandir}/{man1,man5,man8}
-mkdir -p ${RPM_BUILD_ROOT}/run/named
+mkdir -p ${RPM_BUILD_ROOT}/var/run/named
 mkdir -p ${RPM_BUILD_ROOT}/var/log
 
 #chroot
-mkdir -p ${RPM_BUILD_ROOT}/%{chroot_prefix}/{dev,etc,var,run/named}
-mkdir -p ${RPM_BUILD_ROOT}/%{chroot_prefix}/var/{log,named,tmp}
+mkdir -p ${RPM_BUILD_ROOT}/%{chroot_prefix}/{dev,etc,var}
+mkdir -p ${RPM_BUILD_ROOT}/%{chroot_prefix}/var/{log,named,run/named,tmp}
 mkdir -p ${RPM_BUILD_ROOT}/%{chroot_prefix}/etc/crypto-policies/back-ends
 
-# create symlink as it is on real filesystem
-pushd ${RPM_BUILD_ROOT}/%{chroot_prefix}/var
-ln -s ../run run
-popd
-
 mkdir -p ${RPM_BUILD_ROOT}/%{chroot_prefix}/etc/{pki/dnssec-keys,named}
 mkdir -p ${RPM_BUILD_ROOT}/%{chroot_prefix}/%{_libdir}/bind
 # these are required to prevent them being erased during upgrade of previous
@@ -475,54 +453,18 @@
 touch ${RPM_BUILD_ROOT}/%{chroot_prefix}/etc/named.conf
 #end chroot
 
-#sdb-chroot
-%if %{SDB}
-mkdir -p ${RPM_BUILD_ROOT}/%{chroot_sdb_prefix}/{dev,etc,var,run/named}
-mkdir -p ${RPM_BUILD_ROOT}/%{chroot_sdb_prefix}/var/{log,named,tmp}
-mkdir -p ${RPM_BUILD_ROOT}/%{chroot_sdb_prefix}/etc/crypto-policies/back-ends
-
-# create symlink as it is on real filesystem
-pushd ${RPM_BUILD_ROOT}/%{chroot_sdb_prefix}/var
-ln -s ../run run
-popd
-
-mkdir -p ${RPM_BUILD_ROOT}/%{chroot_sdb_prefix}/etc/{pki/dnssec-keys,named}
-mkdir -p ${RPM_BUILD_ROOT}/%{chroot_sdb_prefix}/%{_libdir}/bind
-# these are required to prevent them being erased during upgrade of previous
-touch ${RPM_BUILD_ROOT}/%{chroot_sdb_prefix}/dev/null
-touch ${RPM_BUILD_ROOT}/%{chroot_sdb_prefix}/dev/random
-touch ${RPM_BUILD_ROOT}/%{chroot_sdb_prefix}/dev/zero
-touch ${RPM_BUILD_ROOT}/%{chroot_sdb_prefix}/etc/named.conf
-%endif
-#end sdb-chroot
-
 make DESTDIR=${RPM_BUILD_ROOT} install
 
 # Remove unwanted files
 rm -f ${RPM_BUILD_ROOT}/etc/bind.keys
 
-# Systemd unit files
-mkdir -p ${RPM_BUILD_ROOT}%{_unitdir}
-install -m 644 %{SOURCE37} ${RPM_BUILD_ROOT}%{_unitdir}
-install -m 644 %{SOURCE38} ${RPM_BUILD_ROOT}%{_unitdir}
-install -m 644 %{SOURCE44} ${RPM_BUILD_ROOT}%{_unitdir}
-install -m 644 %{SOURCE46} ${RPM_BUILD_ROOT}%{_unitdir}
-
-%if %{SDB}
-install -m 644 %{SOURCE39} ${RPM_BUILD_ROOT}%{_unitdir}
-install -m 644 %{SOURCE40} ${RPM_BUILD_ROOT}%{_unitdir}
-install -m 644 %{SOURCE45} ${RPM_BUILD_ROOT}%{_unitdir}
-%endif
-
-%if %{PKCS11}
-install -m 644 %{SOURCE47} ${RPM_BUILD_ROOT}%{_unitdir}
-%endif
-
 mkdir -p ${RPM_BUILD_ROOT}%{_libexecdir}
 install -m 755 %{SOURCE41} ${RPM_BUILD_ROOT}%{_libexecdir}/setup-named-chroot.sh
 install -m 755 %{SOURCE42} ${RPM_BUILD_ROOT}%{_libexecdir}/generate-rndc-key.sh
 
+install -m 755 %SOURCE2 ${RPM_BUILD_ROOT}/etc/rc.d/init.d/named
 install -m 644 %SOURCE3 ${RPM_BUILD_ROOT}/etc/logrotate.d/named
+install -m 755 %SOURCE4 ${RPM_BUILD_ROOT}/etc/NetworkManager/dispatcher.d/13-named
 mkdir -p ${RPM_BUILD_ROOT}%{_sysconfdir}/sysconfig
 install -m 644 %{SOURCE1} ${RPM_BUILD_ROOT}%{_sysconfdir}/sysconfig/named
 %if %{SDB}
@@ -578,6 +520,7 @@
 
 # configuration files:
 tar -C ${RPM_BUILD_ROOT} -xjf %{SOURCE28}
+sed -i -e "s@/run/named/@/var/run/named/@g" -e "s@\(include.*bind.config\)@//\1@g" ${RPM_BUILD_ROOT}/etc/named.conf
 touch ${RPM_BUILD_ROOT}/etc/rndc.key
 touch ${RPM_BUILD_ROOT}/etc/rndc.conf
 mkdir ${RPM_BUILD_ROOT}/etc/named
@@ -597,9 +540,6 @@
 done
 :;
 
-mkdir -p ${RPM_BUILD_ROOT}%{_tmpfilesdir}
-install -m 644 %{SOURCE35} ${RPM_BUILD_ROOT}%{_tmpfilesdir}/named.conf
-
 mkdir -p ${RPM_BUILD_ROOT}%{_sysconfdir}/rwtab.d
 install -m 644 %{SOURCE43} ${RPM_BUILD_ROOT}%{_sysconfdir}/rwtab.d/named
 
@@ -612,9 +552,11 @@
 
 %post
 /sbin/ldconfig
-%systemd_post named.service
+/sbin/chkconfig --add named
 if [ "$1" -eq 1 ]; then
-  # Initial installation
+  if [ ! -e /etc/rndc.key ]; then
+    /usr/sbin/rndc-confgen -a -r /dev/urandom> /dev/null 2>&1
+  fi
   [ -x /sbin/restorecon ] && /sbin/restorecon /etc/rndc.* /etc/named.* >/dev/null 2>&1 ;
   # rndc.key has to have correct perms and ownership, CVE-2007-6283
   [ -e /etc/rndc.key ] && chown root:named /etc/rndc.key
@@ -623,40 +565,25 @@
 :;
 
 %preun
-# Package removal, not upgrade
-%systemd_preun named.service
+if [ "$1" -eq 0 ]; then
+  /sbin/service named stop >/dev/null 2>&1 || :;
+  /sbin/chkconfig --del named || :;
+fi;
+:;
 
 %postun
 /sbin/ldconfig
-# Package upgrade, not uninstall
-%systemd_postun_with_restart named.service
+if [ "$1" -ge 1 ]; then
+  /sbin/service named try-restart >/dev/null 2>&1 || :;
+fi;
+:;
 
 %if %{SDB}
 %post sdb
-# Initial installation 
-%systemd_post named-sdb.service
-
-%preun sdb
-# Package removal, not upgrade
-%systemd_preun named-sdb.service
+/sbin/service named try-restart > /dev/null 2>&1 || :;
 
 %postun sdb
-# Package upgrade, not uninstall
-%systemd_postun_with_restart named-sdb.service
-%endif
-
-%if %{PKCS11}
-%post pkcs11
-# Initial installation
-%systemd_post named-pkcs11.service
-
-%preun pkcs11
-# Package removal, not upgrade
-%systemd_preun named-pkcs11.service
-
-%postun pkcs11
-# Package upgrade, not uninstall
-%systemd_postun_with_restart named-pkcs11.service
+/sbin/service named try-restart > /dev/null 2>&1 || :;
 %endif
 
 %triggerpostun -n bind -- bind <= 32:9.5.0-20.b1
@@ -666,10 +593,6 @@
 fi
 :;
 
-%triggerun -- bind < 32:9.9.0-0.6.rc1
-/sbin/chkconfig --del named >/dev/null 2>&1 || :
-/bin/systemctl try-restart named.service >/dev/null 2>&1 || :
-
 %post libs -p /sbin/ldconfig
 
 %postun libs -p /sbin/ldconfig
@@ -684,15 +607,53 @@
 %postun pkcs11-libs -p /sbin/ldconfig
 %endif
 
+# Automatically update configuration from "dnssec-conf-based" to "BIND-based"
+%triggerpostun -n bind -- dnssec-conf
+if [ -r '/etc/named.conf' ]; then
+cp -fp /etc/named.conf /etc/named.conf.rpmsave
+if grep -Eq '/etc/(named.dnssec.keys|pki/dnssec-keys)' /etc/named.conf; then
+  if grep -q 'dlv.isc.org.conf' /etc/named.conf; then
+    # DLV is configured, reconfigure it to new configuration
+    sed -i -e 's/.*dnssec-lookaside.*dlv\.isc\.org\..*/dnssec-lookaside auto;\
+bindkeys-file "\/etc\/named.iscdlv.key";\
+managed-keys-directory "\/var\/named\/dynamic";/' /etc/named.conf
+  fi
+  sed -i -e '/.*named\.dnssec\.keys.*/d' -e '/.*pki\/dnssec-keys.*/d' \
+    /etc/named.conf
+  /sbin/service named try-restart > /dev/null 2>&1 || :;
+fi
+fi
+
+# Ditto for chroot
+if [ -r '/var/named/chroot/etc/named.conf' ]; then
+cp -fp /var/named/chroot/etc/named.conf /var/named/chroot/etc/named.conf.rpmsave
+if grep -Eq '/etc/(named.dnssec.keys|pki/dnssec-keys)' /var/named/chroot/etc/named.conf; then
+  if grep -q 'dlv.isc.org.conf' /var/named/chroot/etc/named.conf; then
+    # DLV is configured, reconfigure it to new configuration
+    sed -i -e 's/.*dnssec-lookaside.*dlv\.isc\.org\..*/dnssec-lookaside auto;\
+bindkeys-file "\/etc\/named.iscdlv.key";\
+managed-keys-directory "\/var\/named\/dynamic";/' /var/named/chroot/etc/named.conf
+  fi
+  sed -i -e '/.*named\.dnssec\.keys.*/d' -e '/.*pki\/dnssec-keys.*/d' \
+    /var/named/chroot/etc/named.conf
+  /sbin/service named try-restart > /dev/null 2>&1 || :;
+fi
+fi
+
 %post chroot
-%systemd_post named-chroot.service
 if [ "$1" -gt 0 ]; then
   [ -e %{chroot_prefix}/dev/random ] || \
     /bin/mknod %{chroot_prefix}/dev/random c 1 8
   [ -e %{chroot_prefix}/dev/zero ] || \
     /bin/mknod %{chroot_prefix}/dev/zero c 1 5
-  [ -e %{chroot_prefix}/dev/null ] || \
+  [ -e %{chroot_prefix}/dev/zero ] || \
     /bin/mknod %{chroot_prefix}/dev/null c 1 3
+  rm -f %{chroot_prefix}/etc/localtime
+  cp /etc/localtime %{chroot_prefix}/etc/localtime
+  if ! grep -q '^ROOTDIR=' /etc/sysconfig/named; then
+    echo 'ROOTDIR=/var/named/chroot' >> /etc/sysconfig/named
+    /sbin/service named try-restart > /dev/null 2>&1 || :;
+  fi
 fi;
 :;
 
@@ -703,52 +664,23 @@
 :;
 
 %preun chroot
-%systemd_preun named-chroot.service 
 if [ "$1" -eq 0 ]; then
-  # Package removal, not upgrade
   rm -f %{chroot_prefix}/dev/{random,zero,null}
+  rm -f %{chroot_prefix}/etc/localtime
+  if grep -q '^ROOTDIR=' /etc/sysconfig/named; then
+    # NOTE: Do NOT call `service named try-restart` because chroot
+    # files will remain mounted.
+    START=no
+    [ -e /var/lock/subsys/named ] && START=yes
+    /sbin/service named stop > /dev/null 2>&1 || :;
+    sed -i -e '/^ROOTDIR=.*/d' /etc/sysconfig/named
+    if [ "x$START" = xyes ]; then
+      /sbin/service named start > /dev/null 2>&1 || :;
+    fi
+  fi
 fi
 :;
 
-%postun chroot
-# Package upgrade, not uninstall
-%systemd_postun_with_restart named-chroot.service
-
-
-%if %{SDB}
-
-%post sdb-chroot
-%systemd_post named-sdb-chroot.service
-if [ "$1" -gt 0 ]; then
-  [ -e %{chroot_sdb_prefix}/dev/random ] || \
-    /bin/mknod %{chroot_sdb_prefix}/dev/random c 1 8
-  [ -e %{chroot_sdb_prefix}/dev/zero ] || \
-    /bin/mknod %{chroot_sdb_prefix}/dev/zero c 1 5
-  [ -e %{chroot_sdb_prefix}/dev/null ] || \
-    /bin/mknod %{chroot_sdb_prefix}/dev/null c 1 3
-fi;
-:;
-
-%posttrans sdb-chroot
-if [ -x /usr/sbin/selinuxenabled ] && /usr/sbin/selinuxenabled; then
-  [ -x /sbin/restorecon ] && /sbin/restorecon %{chroot_sdb_prefix}/dev/* > /dev/null 2>&1;
-fi;
-:;
-
-%preun sdb-chroot
-%systemd_preun named-sdb-chroot.service 
-if [ "$1" -eq 0 ]; then
-  # Package removal, not upgrade
-  rm -f %{chroot_sdb_prefix}/dev/{random,zero,null}
-fi
-:;
-
-%postun sdb-chroot
-# Package upgrade, not uninstall
-%systemd_postun_with_restart named-sdb-chroot.service
-
-%endif
-
 %clean
 rm -rf ${RPM_BUILD_ROOT}
 :;
@@ -759,10 +691,9 @@
 %config(noreplace) %verify(not md5 size mtime) %{_sysconfdir}/sysconfig/named
 %config(noreplace) %attr(0644,root,named) %{_sysconfdir}/named.iscdlv.key
 %config(noreplace) %attr(0644,root,named) %{_sysconfdir}/named.root.key
-%{_tmpfilesdir}/named.conf
+%{_sysconfdir}/rc.d/init.d/named
+%{_sysconfdir}/NetworkManager/dispatcher.d/13-named
 %{_sysconfdir}/rwtab.d/named
-%{_unitdir}/named.service
-%{_unitdir}/named-setup-rndc.service
 %{_sbindir}/named-journalprint
 %{_sbindir}/named-checkconf
 %{_bindir}/named-rrchecker
@@ -810,12 +741,11 @@
 #    so rndc.conf is not necessary.
 %config(noreplace) %{_sysconfdir}/logrotate.d/named
 %defattr(-,named,named,-)
-%dir /run/named
+%dir %{_localstatedir}/run/named
 
 %if %{SDB}
 %files sdb
 %defattr(-,root,root,-)
-%{_unitdir}/named-sdb.service
 %{_mandir}/man1/zone2ldap.1*
 %{_mandir}/man1/ldap2zone.1*
 %{_mandir}/man1/zonetodb.1*
@@ -919,8 +849,6 @@
 
 %files chroot
 %defattr(-,root,root,-)
-%{_unitdir}/named-chroot.service
-%{_unitdir}/named-chroot-setup.service
 %{_libexecdir}/setup-named-chroot.sh
 %ghost %{chroot_prefix}/dev/null
 %ghost %{chroot_prefix}/dev/random
@@ -934,7 +862,7 @@
 %dir %{chroot_prefix}/etc/pki/dnssec-keys
 %dir %{chroot_prefix}/etc/crypto-policies/back-ends
 %dir %{chroot_prefix}/var
-%dir %{chroot_prefix}/run
+%dir %{chroot_prefix}/var/run
 %dir %{chroot_prefix}/var/named
 %ghost %config(noreplace) %{chroot_prefix}/etc/named.conf
 %defattr(-,root,root,-)
@@ -945,47 +873,12 @@
 %dir %{chroot_prefix}/var/tmp
 %dir %{chroot_prefix}/var/log
 %defattr(-,named,named,-)
-%dir %{chroot_prefix}/run/named
-%{chroot_prefix}/var/run
-
-%if %{SDB}
-%files sdb-chroot
-%defattr(-,root,root,-)
-%{_unitdir}/named-sdb-chroot.service
-%{_unitdir}/named-sdb-chroot-setup.service
-%{_libexecdir}/setup-named-chroot.sh
-%ghost %{chroot_sdb_prefix}/dev/null
-%ghost %{chroot_sdb_prefix}/dev/random
-%ghost %{chroot_sdb_prefix}/dev/zero
-%defattr(0640,root,named,0750)
-%dir %{chroot_sdb_prefix}
-%dir %{chroot_sdb_prefix}/dev
-%dir %{chroot_sdb_prefix}/etc
-%dir %{chroot_sdb_prefix}/etc/named
-%dir %{chroot_sdb_prefix}/etc/pki
-%dir %{chroot_sdb_prefix}/etc/pki/dnssec-keys
-%dir %{chroot_sdb_prefix}/etc/crypto-policies/back-ends
-%dir %{chroot_sdb_prefix}/var
-%dir %{chroot_sdb_prefix}/run
-%dir %{chroot_sdb_prefix}/var/named
-%ghost %config(noreplace) %{chroot_sdb_prefix}/etc/named.conf
-%defattr(-,root,root,-)
-%dir %{chroot_sdb_prefix}/usr
-%dir %{chroot_sdb_prefix}/%{_libdir}
-%dir %{chroot_sdb_prefix}/%{_libdir}/bind
-%defattr(0660,named,named,0770)
-%dir %{chroot_sdb_prefix}/var/tmp
-%dir %{chroot_sdb_prefix}/var/log
-%defattr(-,named,named,-)
-%dir %{chroot_sdb_prefix}/run/named
-%{chroot_sdb_prefix}/var/run
-%endif
+%dir %{chroot_prefix}/var/run/named
 
 %if %{PKCS11}
 %files pkcs11
 %defattr(-,root,root,-)
 %{_sbindir}/named-pkcs11
-%{_unitdir}/named-pkcs11.service
 %{_mandir}/man8/named-pkcs11.8*
 
 %files pkcs11-utils
@@ -1012,7 +905,7 @@
 %{_libdir}/libisc-pkcs11.so
 %endif
 
-%files -n python3-bind
+%files -n python34-bind
 %defattr(-,root,root,-)
 %{python3_sitelib}/*py*
 %{python3_sitelib}/isc/*py
