--- gcc.spec.old	2015-09-25 16:18:55.000000000 +0800
+++ gcc.spec	2015-12-07 19:48:24.000000000 +0800
@@ -75,8 +75,8 @@
 %global multilib_32_arch i686
 %endif
 Summary: GCC version 4.9
-Name: %{?scl_prefix}gcc
-#Name: %{?scl_prefix}gcc%{!?scl:49}
+#Name: %{?scl_prefix}gcc
+Name: %{?scl_prefix}gcc%{!?scl:49}
 Version: %{gcc_version}
 Release: %{gcc_release}.2%{?dist}
 # libgcc, libgfortran, libmudflap, libgomp, libstdc++ and crtstuff have
