--- gcc.spec.old	2016-04-08 00:46:20.000000000 +0800
+++ gcc.spec	2017-02-03 15:30:13.000000000 +0800
@@ -79,8 +79,13 @@
 %global multilib_32_arch i686
 %endif
 Summary: GCC version 5
+%if 0%{?moffix} == 6
+#Name: %{?scl_prefix}gcc
+Name: %{?scl_prefix}gcc%{!?scl:5}
+%else
 Name: %{?scl_prefix}gcc
 #Name: %{?scl_prefix}gcc%{!?scl:5}
+%endif
 Version: %{gcc_version}
 Release: %{gcc_release}.1%{?dist}
 # libgcc, libgfortran, libmudflap, libgomp, libstdc++ and crtstuff have
@@ -106,7 +111,12 @@
 # Need binutils which support --build-id >= 2.17.50.0.17-3
 # Need binutils which support %gnu_unique_object >= 2.19.51.0.14
 # Need binutils which support .cfi_sections >= 2.19.51.0.14-33
+%if 0%{?moffix} == 6
+# Need binutils which support Intel MPX >= 2.23.52.0.1-12
+BuildRequires: binutils >= 2.23.52.0.1-12
+%else
 BuildRequires: binutils >= 2.19.51.0.14-33
+%endif
 # While gcc doesn't include statically linked binaries, during testing
 # -static is used several times.
 BuildRequires: glibc-static
@@ -163,8 +173,13 @@
 Requires: %{?scl_prefix}binutils >= 2.22.52.0.1
 %endif
 %else
+%if 0%{?moffix} == 6
+# Need binutils which support Intel MPX >= 2.23.52.0.1-12
+Requires: binutils >= 2.23.52.0.1-12
+%else
 Requires: binutils >= 2.19.51.0.14-33
 %endif
+%endif
 # Make sure gdb will understand DW_FORM_strp
 Conflicts: gdb < 5.1-2
 Requires: glibc-devel >= 2.2.90-12
@@ -1704,7 +1719,11 @@
 cd ..
 
 %if 0%{!?scl:1}
+%if 0%{?moffix} == 6
+for i in %{buildroot}%{_prefix}/bin/{*gcc,*++,gcov*,gfortran,gcc-ar,gcc-nm,gcc-ranlib}; do
+%else
 for i in %{buildroot}%{_prefix}/bin/{*gcc,*++,gcov,gfortran,gcc-ar,gcc-nm,gcc-ranlib}; do
+%endif
   mv -f $i ${i}5
 done
 %endif
