--- gmp.spec.old	2012-06-26 17:40:21.000000000 +0800
+++ gmp.spec	2017-01-18 09:31:18.000000000 +0800
@@ -13,6 +13,10 @@
 Source2: gmp.h
 Source3: gmp-mparam.h
 Patch0: gmp-4.0.1-s390.patch
+%if 0%{?moffix} == 6
+Patch1: gmp-4.3.2-t-scan.patch
+Patch2: gmp-4.3.2-ansi2knr.patch
+%endif
 License: LGPLv3+
 Group: System Environment/Libraries
 BuildRoot: %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
@@ -57,6 +61,10 @@
 %prep
 %setup -q
 %patch0 -p1 -b .s390
+%if 0%{?moffix} == 6
+%patch1 -p1 -b .t-scan
+%patch2 -p1 -b .ansi2knr
+%endif
 
 %build
 autoreconf -if
