--- compat-icu.spec.old	2016-03-03 00:31:58.000000000 +0800
+++ compat-icu.spec	2017-01-27 19:21:22.000000000 +0800
@@ -1,4 +1,8 @@
+%if 0%{?moffix} == 6
+Name:      compat-icu
+%else
 Name:      icu
+%endif
 Version:   4.2.1
 Release:   14%{?dist}
 Summary:   International Components for Unicode
@@ -25,11 +29,19 @@
 %description
 Tools and utilities for developing with icu.
 
+%if 0%{?moffix} == 6
+%package -n compat-libicu
+%else
 %package -n lib%{name}
+%endif
 Summary: International Components for Unicode - libraries
 Group:   System Environment/Libraries
 
+%if 0%{?moffix} == 6
+%description -n compat-libicu
+%else
 %description -n lib%{name}
+%endif
 The International Components for Unicode (ICU) libraries provide
 robust and full-featured Unicode services on a wide variety of
 platforms. ICU supports the most current version of the Unicode
@@ -41,6 +53,7 @@
 sacrificing performance. It offers great flexibility to extend and
 customize the supplied services.
 
+%if 0%{?moffix} != 6
 %package  -n lib%{name}-devel
 Summary:  Development files for International Components for Unicode
 Group:    Development/Libraries
@@ -57,9 +70,14 @@
 
 %description -n lib%{name}-doc
 %{summary}.
+%endif
 
 %prep
+%if 0%{?moffix} == 6
+%setup -q -n icu
+%else
 %setup -q -n %{name}
+%endif
 %patch1 -p1 -b .multiarchdevel
 %patch2 -p1 -b .icu6995.kannada.patch
 %patch3 -p1 -b .icu7039.badextract.patch
@@ -83,6 +101,11 @@
 make doc
 
 %install
+%if 0%{?moffix} == 6
+rm -rf $RPM_BUILD_ROOT
+mkdir -p $RPM_BUILD_ROOT%{_libdir}
+cp -a source/lib/*.so.* $RPM_BUILD_ROOT%{_libdir}/
+%else
 rm -rf $RPM_BUILD_ROOT source/__docs
 make -C source install DESTDIR=$RPM_BUILD_ROOT
 make -C source install-doc docdir=__docs
@@ -91,6 +114,7 @@
 chmod 0755 $RPM_BUILD_ROOT%{_bindir}/%{name}-config
 sed -i s/\\\$\(THREADSCXXFLAGS\)// $RPM_BUILD_ROOT/%{_libdir}/pkgconfig/icu.pc
 sed -i s/\\\$\(THREADSCPPFLAGS\)/-D_REENTRANT/ $RPM_BUILD_ROOT/%{_libdir}/pkgconfig/icu.pc
+%endif
 
 %check
 make -C source check
@@ -98,10 +122,17 @@
 %clean
 rm -rf $RPM_BUILD_ROOT
 
+%if 0%{?moffix} == 6
+%post -n compat-libicu -p /sbin/ldconfig
+
+%postun -n compat-libicu -p /sbin/ldconfig
+%else
 %post -n lib%{name} -p /sbin/ldconfig
 
 %postun -n lib%{name} -p /sbin/ldconfig
+%endif
 
+%if 0%{?moffix} != 6
 %files
 %defattr(-,root,root,-)
 %doc license.html readme.html
@@ -124,11 +155,17 @@
 %{_mandir}/man1/pkgdata.1*
 %{_mandir}/man1/uconv.1*
 %{_mandir}/man8/*.8*
+%endif
 
+%if 0%{?moffix} == 6
+%files -n compat-libicu
+%else
 %files -n lib%{name}
+%endif
 %defattr(-,root,root,-)
 %{_libdir}/*.so.*
 
+%if 0%{?moffix} != 6
 %files -n lib%{name}-devel
 %defattr(-,root,root,-)
 %{_bindir}/%{name}-config
@@ -148,6 +185,7 @@
 %files -n lib%{name}-doc
 %defattr(-,root,root,-)
 %doc source/__docs/%{name}/html/*
+%endif
 
 %changelog
 * Wed Mar 02 2016 Eike Rathke <erack@redhat.com> - 4.2.1-14
