--- cmake.spec.old	2016-10-03 15:38:25.000000000 +0800
+++ cmake.spec	2017-01-17 16:32:49.000000000 +0800
@@ -82,6 +82,10 @@
 %if 0%{?rhel} && 0%{?rhel} <= 6
 Patch1001:      %{name}-libarchive3.patch
 %endif
+%else
+%if 0%{?moffix} == 6
+Patch1001:      %{name}3-libarchive3.patch
+%endif
 %endif
 
 BuildRequires:  gcc-gfortran
@@ -182,6 +186,10 @@
 %if 0%{?rhel} && 0%{?rhel} <= 6
 %patch1001 -p1
 %endif
+%else
+%if 0%{?moffix} == 6
+%patch1001 -p1
+%endif
 %endif
 
 # We cannot use backups with patches to Modules as they end up being installed
