--- cmake.spec.old	2017-09-21 16:09:20.000000000 +0800
+++ cmake.spec	2017-10-27 09:29:08.000000000 +0800
@@ -7,7 +7,11 @@
 
 # Set to bcond_without or use --with bootstrap if bootstrapping a new release
 # or architecture
+%if 0%{?moffix} == 6
+%bcond_without bootstrap
+%else
 %bcond_with bootstrap
+%endif
 
 # Build with Emacs support
 %bcond_without cmake_enables_emacs
@@ -379,6 +383,9 @@
 %ifarch s390x
 NO_TEST="$NO_TEST|RunCMake.File_Generate"
 %endif
+%if 0%{?moffix} == 6
+NO_TEST="$NO_TEST|Server"
+%endif
 export NO_TEST
 bin/ctest%{?name_suffix} -V -E "$NO_TEST" %{?_smp_mflags}
 popd
