--- cmake.spec.old	2016-11-12 22:39:55.000000000 +0800
+++ cmake.spec	2016-12-16 14:38:20.000000000 +0800
@@ -1,6 +1,6 @@
 # Set to bcond_without or use --with bootstrap if bootstrapping a new release
 # or architecture
-%bcond_with bootstrap
+%bcond_without bootstrap
 
 # Set to bcond_with or use --without gui to disable qt4 gui build
 %bcond_without gui
@@ -77,19 +77,19 @@
 
 BuildRequires:  gcc-gfortran
 BuildRequires:  ncurses-devel, libX11-devel
-BuildRequires:  bzip2-devel
-BuildRequires:  curl-devel
-BuildRequires:  expat-devel
-BuildRequires:  jsoncpp-devel
+#BuildRequires:  bzip2-devel
+#BuildRequires:  curl-devel
+#BuildRequires:  expat-devel
+#BuildRequires:  jsoncpp-devel
 %if 0%{?fedora} || 0%{?rhel} >= 7
-BuildRequires:  libarchive-devel
+#BuildRequires:  libarchive-devel
 BuildRequires:  /usr/bin/sphinx-build
 %else
-BuildRequires:  libarchive3-devel
+#BuildRequires:  libarchive3-devel
 %endif
-BuildRequires:  libuv-devel
-BuildRequires:  xz-devel
-BuildRequires:  zlib-devel
+#BuildRequires:  libuv-devel
+#BuildRequires:  xz-devel
+#BuildRequires:  zlib-devel
 BuildRequires:  emacs
 %if %{with python3}
 %{!?python3_pkgversion: %global python3_pkgversion 3}
