--- openvas-scanner.spec.old	2016-11-14 03:30:03.000000000 +0800
+++ openvas-scanner.spec	2016-12-06 09:56:43.000000000 +0800
@@ -28,6 +28,7 @@
 # Replace _BSD_SOURCE and _SVID_SOURCE with _DEFAULT_SOURCE otherwise build fails with Werror
 Patch4:		openvas-scanner-bsdsource.patch
 
+Patch5:         openvas-scanner-gcc44-strict-aliasing.patch
 
 BuildRequires:	openvas-libraries-devel >= 7.0
 BuildRequires:	cmake >= 2.6.0
@@ -89,6 +90,8 @@
 #%patch4 -p 1 -b .bsdsource
 #%endif
 
+%patch5 -p1 -b .strict-aliasing
+
 for i in CHANGES ChangeLog; do
 	iconv -f iso8859-1 -t utf-8 $i > $i.utf8 && \
 	touch -r $i $i.utf8 && \
