--- compat-ImageMagick.spec.old	2016-06-03 23:11:39.000000000 +0800
+++ compat-ImageMagick.spec	2017-01-27 17:13:31.000000000 +0800
@@ -1,16 +1,26 @@
 %define VER 6.7.2
 %define Patchlevel 7
 
+%if 0%{?moffix} == 6
+Name:           compat-ImageMagick
+%else
 Name:           ImageMagick
+%endif
 Version:        %{VER}.%{Patchlevel}
 Release:        5%{?dist}
 Summary:        An X application for displaying and manipulating images
 Group:          Applications/Multimedia
 License:        ImageMagick
 Url:            http://www.imagemagick.org/
+%if 0%{?moffix} == 6
+# Source0:        ftp://ftp.ImageMagick.org/pub/ImageMagick/ImageMagick-%{VER}-%{Patchlevel}.tar.lzma
+# Reference mirror ftp://ftp.nluug.nl instead which keeps older releases longer
+Source0:        ftp://ftp.nluug.nl/pub/ImageMagick/ImageMagick-%{VER}-%{Patchlevel}.tar.gz
+%else
 # Source0:        ftp://ftp.ImageMagick.org/pub/%{name}/%{name}-%{VER}-%{Patchlevel}.tar.lzma
 # Reference mirror ftp://ftp.nluug.nl instead which keeps older releases longer
 Source0:        ftp://ftp.nluug.nl/pub/%{name}/%{name}-%{VER}-%{Patchlevel}.tar.gz
+%endif
 Patch1:         ImageMagick-6.4.0-multilib.patch
 # CVE-2012-0247 CVE-2012-0248 CVE-2012-1185 CVE-2012-1186
 Patch3: ImageMagick-6.7.2-cve-2012-0247-0248.patch
@@ -50,6 +60,7 @@
 which use ImageMagick code or APIs, you need to install
 ImageMagick-devel as well.
 
+%if 0%{?moffix} != 6
 %package devel
 Summary: Library links and header files for ImageMagick app development
 Group: Development/Libraries
@@ -97,6 +108,7 @@
 
 Install ImageMagick-perl if you want to use any perl scripts that use
 ImageMagick.
+%endif
 
 
 %package c++
@@ -111,6 +123,7 @@
 Install ImageMagick-c++ if you want to use any applications that use Magick++.
 
 
+%if 0%{?moffix} != 6
 %package c++-devel
 Summary: C++ bindings for the ImageMagick library
 Group: Development/Libraries
@@ -128,10 +141,15 @@
 You don't need to install it if you just want to use ImageMagick, or if you
 want to develop/compile applications using the ImageMagick C interface,
 however.
+%endif
 
 
 %prep
+%if 0%{?moffix} == 6
+%setup -q -n ImageMagick-%{VER}-%{Patchlevel}
+%else
 %setup -q -n %{name}-%{VER}-%{Patchlevel}
+%endif
 %patch1 -p1 -b .multilib
 %patch3 -p1 -b .cve-2012-0247-0248
 %patch4 -p1 -b .cve-2012-0259-0260
@@ -183,7 +201,11 @@
 rm -rf $RPM_BUILD_ROOT
 
 make install DESTDIR=$RPM_BUILD_ROOT INSTALL="install -p"
+%if 0%{?moffix} == 6
+cp -a www/source $RPM_BUILD_ROOT%{_datadir}/doc/ImageMagick-%{VER}
+%else
 cp -a www/source $RPM_BUILD_ROOT%{_datadir}/doc/%{name}-%{VER}
+%endif
 rm $RPM_BUILD_ROOT%{_libdir}/*.la
 
 # fix weird perl Magick.so permissions
@@ -217,10 +239,17 @@
 %define wordsize 32
 %endif
 
+%if 0%{?moffix} == 6
+mv $RPM_BUILD_ROOT%{_includedir}/ImageMagick/magick/magick-config.h \
+   $RPM_BUILD_ROOT%{_includedir}/ImageMagick/magick/magick-config-%{wordsize}.h
+
+cat >$RPM_BUILD_ROOT%{_includedir}/ImageMagick/magick/magick-config.h <<EOF
+%else
 mv $RPM_BUILD_ROOT%{_includedir}/%{name}/magick/magick-config.h \
    $RPM_BUILD_ROOT%{_includedir}/%{name}/magick/magick-config-%{wordsize}.h
 
 cat >$RPM_BUILD_ROOT%{_includedir}/%{name}/magick/magick-config.h <<EOF
+%endif
 #ifndef IMAGEMAGICK_MULTILIB
 #define IMAGEMAGICK_MULTILIB
 
@@ -237,6 +266,16 @@
 #endif
 EOF
 
+%if 0%{?moffix} == 6
+rm -f $RPM_BUILD_ROOT%{_bindir}/*
+rm -fr $RPM_BUILD_ROOT%{_mandir}
+rm -f $RPM_BUILD_ROOT%{_libdir}/*.so
+rm -fr $RPM_BUILD_ROOT%{_libdir}/pkgconfig
+rm -fr $RPM_BUILD_ROOT%{_libdir}/perl5
+rm -fr $RPM_BUILD_ROOT%{_includedir}
+rm -fr $RPM_BUILD_ROOT%{_datadir}/doc
+%endif
+
 %clean
 rm -rf $RPM_BUILD_ROOT
 
@@ -256,13 +295,20 @@
 %doc README.txt LICENSE NOTICE AUTHORS.txt NEWS.txt
 %{_libdir}/libMagickCore.so.*
 %{_libdir}/libMagickWand.so.*
+%if 0%{?moffix} == 6
+%{_libdir}/ImageMagick-%{VER}
+%{_datadir}/ImageMagick-%{VER}
+%{_sysconfdir}/ImageMagick
+%else
 %{_bindir}/[a-z]*
 %{_libdir}/%{name}-%{VER}
 %{_datadir}/%{name}-%{VER}
 %{_mandir}/man[145]/[a-z]*
 %{_mandir}/man1/%{name}.*
 %{_sysconfdir}/%{name}
+%endif
 
+%if 0%{?moffix} != 6
 %files devel
 %defattr(-,root,root,-)
 %{_bindir}/MagickCore-config
@@ -286,6 +332,7 @@
 %files doc
 %defattr(-,root,root,-)
 %doc %{_datadir}/doc/%{name}-%{VER}
+%endif
 
 %files c++
 %defattr(-,root,root,-)
@@ -293,6 +340,7 @@
 %doc www/Magick++/COPYING
 %{_libdir}/libMagick++.so.*
 
+%if 0%{?moffix} != 6
 %files c++-devel
 %defattr(-,root,root,-)
 %doc Magick++/examples
@@ -308,6 +356,7 @@
 %defattr(-,root,root,-)
 %{_mandir}/man3/*
 %doc PerlMagick/demo/ PerlMagick/Changelog PerlMagick/README.txt
+%endif
 
 
 %changelog
